packages/main/src/plugin/kubernetes/contexts-preflight-checker.ts

```diff
@@ -0,0 +1,112 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import path from 'node:path';
+
+import { isWindows } from '/@/util.js';
+
+import type { Exec } from '../util/exec.js';
+import type { KubeConfigSingleContext } from './kubeconfig-single-context.js';
+
+export interface PreflightCheckContextResults {
+  contextName: string;
+  results: PreflightCheckResult[];
+}
+
+export interface PreflightCheckResult {
+  description: string;
+}
+
+export class ContextsPreflightChecker {
```

Comment:
Will this class manage more than the [kubeconfig.v1/#ExecConfig](https://kubernetes.io/docs/reference/config-api/kubeconfig.v1/#ExecConfig) ?

Is it planned to run the command defined in the `ExecConfig` later on to refresh the connections?

Reply:
The idea of this class is to run any kind of check, to analyze the kubeconfig and detect some problems